axelor-docusign/src/main/java/com/axelor/apps/docusign/web/DocuSignEnvelopeController.java

```diff
@@ -0,0 +1,112 @@
+
+    Query<DocuSignEnvelopeSetting> envelopeSettingQuery =
+        Beans.get(DocuSignEnvelopeSettingRepository.class)
+            .all()
+            .filter("self.metaModel.fullName = ?", model);
+
+    try {
+      long settingsCount = envelopeSettingQuery.count();
+
+      if (settingsCount == 0) {
+        response.setView(
+            ActionView.define(I18n.get("Create envelope"))
+                .model(DocuSignEnvelope.class.getName())
+                .add("form", "docusign-envelope-form")
+                .param("forceEdit", "true")
+                .context("_templateContextModel", model)
+                .context("_objectId", context.getId().toString())
+                .map());
+      } else if (settingsCount == 1) {
+        response.setView(
+            Beans.get(DocuSignEnvelopeService.class)
+                .generateEnvelope(envelopeSettingQuery.fetchOne(), context.getId()));
+      } else if (settingsCount >= 2) {
+        response.setView(
+            ActionView.define(I18n.get("Select envelope setting"))
+                .model(Wizard.class.getName())
+                .add("form", "docusign-select-envelope-setting-wizard-form")
+                .param("show-confirm", "false")
+                .context("_objectId", context.getId().toString())
+                .context("_templateContextModel", model)
+                .context("_simpleModel", simpleModel)
+                .map());
+      }
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void createEnvelopeFromWizard(ActionRequest request, ActionResponse response) {
+    Context context = request.getContext();
+
+    Map envelopeSettingContext = (Map) context.get("envelopeSetting");
+    DocuSignEnvelopeSetting envelopeSetting;
+    if (MapUtils.isNotEmpty(envelopeSettingContext)) {
+      envelopeSetting =
+          Beans.get(DocuSignEnvelopeSettingRepository.class)
+              .find(Long.parseLong(envelopeSettingContext.get("id").toString()));
+
+      Long objectId = Long.parseLong(context.get("_objectId").toString());
+      String model = (String) context.get("_templateContextModel");
+      String simpleModel = (String) context.get("_simpleModel");
+
+      try {
+        response.setView(
+            Beans.get(DocuSignEnvelopeService.class).generateEnvelope(envelopeSetting, objectId));
+      } catch (Exception e) {
+        TraceBackService.trace(response, e);
+      }
+    }
+  }
+
+  public void sendEnvelope(ActionRequest request, ActionResponse response) {
+    try {
+      DocuSignEnvelope envelope = request.getContext().asType(DocuSignEnvelope.class);
+      envelope = Beans.get(DocuSignEnvelopeRepository.class).find(envelope.getId());
+      Beans.get(DocuSignEnvelopeService.class).sendEnvelope(envelope);
+      response.setReload(true);
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+      response.setError(I18n.get(e.getMessage()));
```

Comment:
You can call `TraceBackService.trace(response, e, ResponseMessageType.ERROR);` instead to avoid having multiple popup